fast_api_proj/app/main.py

```diff
@@ -0,0 +1,78 @@
+from fastapi import FastAPI, HTTPException, Depends
+from app.models.user import User, AdminUser, RegularUser
+from app.dependecies import (
+    get_user_service,
+    get_admin_user_service,
+    get_regular_user_service,
+    get_user_retrieval_service,
+    get_user_profile_update_service,
+    delete_user_service
+)
+from app.config import config
+from app.services.in_memory_user_service import InMemoryUserService
+
+app = FastAPI()
+
+
+@app.post("/user/")
+def create_user(user: User, service=Depends(get_user_service)):
+    try:
+        return service.create_user(user)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+
+
+@app.post("/admin_user/")
+def create_admin_user(user: AdminUser, service=Depends(get_admin_user_service)):
+    # Check max admin users limit
+    if isinstance(service, InMemoryUserService):
+        current_admins = [key for key in service.users if isinstance(service.users[key], AdminUser)]
+    else:
+        current_admins = [key for key in service.redis_client.keys() if key.startswith('admin')]
+
+    if len(current_admins) >= config.MAX_ADMIN_USERS:
+        raise HTTPException(status_code=400, detail="Maximum number of admin users reached")
+
+    try:
+        return service.create_user(user)
+    except ValueError as e:
```

Comment:
Why is a `ValueError` necessarily a status code 400? maybe it's an internal server error? If u want to dedicate specific errors that are caused by the user, they should raise their own exception which you can catch.
This is relevant for all the `except ValueError as e:`